writer.go

```diff
@@ -57,28 +57,30 @@ type multiLevelWriter struct {
 func (t multiLevelWriter) Write(p []byte) (n int, err error) {
 	for _, w := range t.writers {
 		n, err = w.Write(p)
-		if err != nil {
-			return
-		}
 		if n != len(p) {
 			err = io.ErrShortWrite
-			return
 		}
 	}
+
+	if err != nil {
```

Comment:
This won't work, as if the last writer does not have an error, err will be overwritten with `nil`.